File: src/main/DataProvider/DataProvider/DataProvider.java
package DataProvider;

import Database.DatabaseGraph;
import Query.QueryGraph;
import Transducer.TransducerGraph;

import java.util.HashSet;
import java.util.Set;

// the dataProvider holds the three data graphs.

public class DataProvider {

    public QueryGraph queryGraph = new QueryGraph();
    public TransducerGraph transducerGraph = new TransducerGraph();
    public DatabaseGraph databaseGraph = new DatabaseGraph();
    public Set<String> alphabet;

    public String getDataSetIdentifier() {
        return dataSetIdentifier;
    }

    public void setDataSetIdentifier(String dataSetIdentifier) {
        this.dataSetIdentifier = dataSetIdentifier;
    }

    public String dataSetIdentifier;


    public DataProvider() {
        QueryGraph queryGraph = new QueryGraph();
        TransducerGraph transducerGraph = new TransducerGraph();
        DatabaseGraph databaseGraph = new DatabaseGraph();
        alphabet = new HashSet<>();
    }

    public QueryGraph getQueryGraph() {
        return queryGraph;
    }

    public TransducerGraph getTransducerGraph() {
        return transducerGraph;
    }

    public DatabaseGraph getDatabaseGraph() {
        return databaseGraph;
    }


}

File: src/main/kotlin/tin/model/tintheweb/DataProvider.kt
package tin.model.tintheweb


import tin.model.database.DatabaseGraph
import tin.model.query.QueryGraph
import tin.model.transducer.TransducerGraph
import java.util.HashSet

// the dataProvider holds the three data graphs.
class DataProvider(
        var queryGraph: QueryGraph,
        var transducerGraph: TransducerGraph,
        var databaseGraph: DatabaseGraph,
        var alphabet: Set<String> = HashSet()
) {

    var dataSetIdentifier: String? = null

}